pkgs/by-name/ha/handheld-daemon/package.nix

```
@@ -1,63 +1,90 @@
 {
+  lib,
+  python3Packages,
   fetchFromGitHub,
+
+  # dependencies
+  systemd,
   hidapi,
+  coreutils,
   kmod,
-  lib,
-  python3,
-  toybox,
-  lsof
+  dbus,
+  lsof,
 }:
-python3.pkgs.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
   pname = "handheld-daemon";
-  version = "3.6.1";
+  version = "3.7.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "hhd-dev";
     repo = "hhd";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-IdpSRb66G+WzTv/BC29r2OjO1b4VdWbV6OSzOoiFAO0=";
+    tag = "v${version}";
+    hash = "sha256-DkVdYnSEeaNZj76lhdU+9Pl0yzam2A2QGa3aHCmSHEA=";
   };
 
-  propagatedBuildInputs = with python3.pkgs; [
+  # This package relies on several programs expected to be on the user's PATH.
+  # We take a more reproducible approach by patching the absolute path to each of these required
+  # binaries.
+  postPatch = ''
+    substituteInPlace src/hhd/controller/lib/hid.py \
+      --replace-fail "libhidapi" "${lib.getLib hidapi}/lib/libhidapi"
+
+    substituteInPlace src/hhd/controller/lib/hide.py \
+      --replace-fail "/bin/chmod" "${lib.getExe' coreutils "chmod"}" \
+      --replace-fail "udevadm" "${lib.getExe' systemd "udevadm"}"
+
+    substituteInPlace src/hhd/controller/physical/evdev.py \
+      --replace-fail "udevadm" "${lib.getExe' systemd "udevadm"}"
+
+    substituteInPlace src/hhd/controller/physical/imu.py \
+      --replace-fail '"modprobe' '"${lib.getExe' kmod "modprobe"}'
+
+    substituteInPlace src/hhd/device/oxp/serial.py \
+      --replace-fail "udevadm" "${lib.getExe' systemd "udevadm"}"
+
+    substituteInPlace src/hhd/plugins/overlay/systemd.py \
+      --replace-fail "dbus-monitor" "${lib.getExe' dbus "dbus-monitor"}" \
+      --replace-fail "systemd-inhibit" "${lib.getExe' systemd "systemd-inhibit"}"
+
+    substituteInPlace src/hhd/plugins/overlay/x11.py \
+      --replace-fail "lsof" "${lib.getExe lsof}"
+
+    substituteInPlace src/hhd/plugins/plugin.py \
+      --replace-fail '"id"' '"${lib.getExe' coreutils "id"}"'
+  '';
```

```suggestion
    substituteInPlace src/hhd/plugins/plugin.py \
      --replace-fail '"id"' '"${lib.getExe' coreutils "id"}"'
    substituteInPlace src/hhd/plugins/overlay/power.py \
      --replace-fail '"efibootmgr"' '"${lib.getExe' efibootmgr "id"}"'
  '';
```

[This file](https://github.com/hhd-dev/hhd/blob/c664e1530301a893648e6b09b6d1bc7b41568197/src/hhd/plugins/overlay/power.py) uses both efibootmgr and systemctl. Only suggesting efibootctl since systemctl seems to already be in `handheld-daemon`'s path